doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/index.md

```diff
@@ -5,4 +5,366 @@ permalink: /:path/
+
+```
+CMakeLists.txt
+driver/
+    uart.h
+    uart.c
+ld/
+    eagle.app.v6.ld
+    eagle.rom.addr.v6.ld
+kaa/
+    <put Kaa SDK here>
+user/
+    user_main.c
+src/
+    kaa_demo.c
+```
+
+Some notes:
+
+* `CMakeLists.txt` is a CMake script (see below)
+* `driver/uart.c` and `driver/uart.h` files implement driver for ESP8266 UART interface.
+* The `ld/` directory contains two linker scripts required for ESP8266 applications.
+* You should put generated Kaa C SDK tarball into `kaa/` directory and unpack it:
+
+        cd kaa && tar zxf kaa-c*.tar.gz
+* `user/user_main.c` contains ESP8266 application entry ponit (`user_init()` function)
+and performs ESP8266-specific initizalizations (e.g. initialize UART).
+* `src/kaa_demo.c` is a platofrm-independent source file with minimal Kaa code.
+
+### Minimal code
+
+Each ESP8266 application starts its execution form `user_init()` function.
+Here we initialize UART and set it's baudrate to 115200 baud:
+
+```c
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "uart.h"
+
+void user_init(void)
+{
+    uart_init_new();
+    UART_SetBaudrate(UART0, 115200);
+    UART_SetPrintPort(UART0);
+```
+
+Next, we should start a system task in `user_init()`, since we run in FreeRTOS environment:
+
+```c
+    portBASE_TYPE error = xTaskCreate(main_task, "main_task",
+            512, NULL, 2, NULL );
+    if (error < 0) {
+        printf("Error creating main_task! Error code: %ld\r\n", error);
+    }
+}
+```
+
+In scope of this example, `main_task()` just calls `main()` function:
+
+```c
+static void main_task(void *pvParameters)
+{
+    (void)pvParameters;
+    main();
+    for (;;);
+}
+```
+
+The `main()` function is defined in `src/kaa_demo.c` and starts minimal Kaa client using Kaa C SDK:
+
+```c
+#include <stddef.h>
+#include <kaa/kaa_error.h>
+#include <kaa/kaa_context.h>
+#include <kaa/platform/kaa_client.h>
+
+static void loop_fn(void *context)
+{
+    printf("Hello, Kaa!\r\n");
+
+    kaa_client_stop(context);
+}
+
+int main(void)
+{
+    printf("Initializing Kaa client\r\n");
+
+    kaa_client_t *kaa_client;
+
+    kaa_error_t error_code = kaa_client_create(&kaa_client, NULL);
+    if (error_code) {
+        printf("Failed to create Kaa client\r\n");
+        return 1;
+    }
+
+    error_code = kaa_client_start(kaa_client, loop_fn, kaa_client, 0);
+    if (error_code) {
+        printf("Failed to start Kaa main loop\r\n");
+        return 1;
+    }
+
+    kaa_client_destroy(kaa_client);
+    return 0;
+}
+```
+
+### Build system overview
+
+The Kaa C SDK makes use of CMake build system generator. Although you can choose any build system of your preference, it is recommended to use CMake for Kaa applications as well.
+That will make possible to tightly integrate your application's build system with Kaa SDK and use already provided ESP8266 toolchain file.
+
+#### CMakeLists.txt
+
+First, let's tell CMake minimum version required and project name.
+```CMake
```

Comment:
What do you mean by "code style"? If you mean "code coloring", that's not PR's issue, but our Jekyll setup in general.